packages/inferno-router/src/resolveLoaders.ts

```diff
@@ -0,0 +1,198 @@
+import { isNullOrUndef, isUndefined } from "inferno-shared";
+import { matchPath } from "./matchPath";
+import type { TLoaderData, TLoaderProps } from "./Router";
+import { Switch } from "./Switch";
+
+export function resolveLoaders(loaderEntries: TLoaderEntry[]): Promise<Record<string, TLoaderData>> {
+  const promises = loaderEntries.map((({ path, params, request, loader }) => {
+    return resolveEntry(path, params, request, loader);
+  }));
+  return Promise.all(promises).then((result) => {
+    return Object.fromEntries(result);
+  });
+}
+
+type TLoaderEntry = {
+  path: string,
+  params: Record<string, any>,
+  request: Request,
+  controller: AbortController,
+  loader: (TLoaderProps) => Promise<TLoaderEntry>,
+}
+
+// Optionally pass base param during SSR to get fully qualified request URI passed to loader in request param
+export function traverseLoaders(location: string, tree: any, base?: string, parentIsSwitch = false): TLoaderEntry[] {
+  // Make sure tree isn't null
+  if (isNullOrUndef(tree)) return [];
+
+  if (Array.isArray(tree)) {
+    let hasMatch = false;
+    const entries = tree.reduce((res, node) => {
+      if (parentIsSwitch && hasMatch) return res;
+
+      const outpArr = traverseLoaders(location, node, base, node?.type === Switch);
```

Comment:
Switch is exposed and not sealed, so one can implement their own switch, I believe we should check for inheritance also here

Reply:
WILL FIX

Reply:
Fixed in b79c8f9e169

Reply:
Here is still === Switch comparison

Reply:
Fixed in  e7f33a97d (and b79c8f9)